air/src/aux_table/bitwise_pow2/tests.rs

```diff
@@ -1,12 +1,68 @@
-use super::{bitwise, pow2, PERIODIC_CYCLE_LEN};
+use super::bitwise::{NUM_CONSTRAINTS, NUM_DECOMP_BITS};
+use super::{bitwise, pow2, EvaluationFrame, BITWISE_TRACE_OFFSET, PERIODIC_CYCLE_LEN};
 use crate::{Felt, FieldElement};
-use vm_core::bitwise::{Selectors, BITWISE_AND, BITWISE_OR, BITWISE_XOR};
+use vm_core::{
+    bitwise::{
+        Selectors, BITWISE_AND, BITWISE_OR, BITWISE_XOR, NUM_SELECTORS,
+        TRACE_WIDTH as BITWISE_TRACE_WIDTH,
+    },
+    TRACE_WIDTH,
+};
 
 use proptest::prelude::*;
 
 // UNIT TESTS
 // ================================================================================================
 
+/// Takes two custom consecutive row cycle state of bitwise decompostion with two distinct selectors. Outputs failure
+/// as all constraint are not zero (not satisfied)
+#[test]
+fn test_bitwise_selectors_fail() {
+    let current_bitwise = vec![
+        // decomposed a
+        Felt::ONE,
+        Felt::ONE,
+        Felt::ZERO,
+        Felt::ZERO,
+        // decomposed b
+        Felt::ONE,
+        Felt::ZERO,
+        Felt::ONE,
+        Felt::ONE,
+    ];
+
+    let next_bitwise = vec![
+        Felt::ONE,
+        Felt::ONE,
+        Felt::ONE,
+        Felt::ZERO,
+        Felt::ZERO,
+        Felt::ONE,
+        Felt::ONE,
+        Felt::ONE,
+    ];
+    let cycle = 1;
+
+    let frame =
+        get_test_frame_with_two_selectors(&current_bitwise, &next_bitwise, BITWISE_AND, BITWISE_OR);
+    let (result, expected) = get_bitwise_frame_result(frame, cycle);
+    assert_ne!(result, expected);
+
+    let frame = get_test_frame_with_two_selectors(
+        &current_bitwise,
+        &next_bitwise,
+        BITWISE_XOR,
+        BITWISE_AND,
+    );
+    let (result, expected) = get_bitwise_frame_result(frame, cycle);
+    assert_ne!(result, expected);
+
+    let frame =
+        get_test_frame_with_two_selectors(&current_bitwise, &next_bitwise, BITWISE_OR, BITWISE_XOR);
+    let (result, expected) = get_bitwise_frame_result(frame, cycle);
+    assert_ne!(result, expected);
```

Comment:
With `assert_ne` at this level, I would prefer to define `expected` directly in this function and only return `result` from `get_bitwise_frame_result`. Since it's returning the same `expected` value (zeros) every time and that value doesn't depend on anything in the function, there's no need to have it come the function, and it could be defined just one time at this level.